tests/listobjects/listobjects.go

```diff
@@ -108,19 +109,31 @@ func runTests(t *testing.T, schemaVersion string, client ListObjectsClientInterf
 					require.NoError(t, err)
 				}
 
-				for _, assertion := range stage.Assertions {
+				for _, assertion := range stage.ListAssertions {
 					// assert 1: on regular list objects endpoint
+
+					var contextualTuples *pb.ContextualTupleKeys
+					if len(assertion.ContextualTuples) > 0 {
+						contextualTuples = &pb.ContextualTupleKeys{
+							TupleKeys: assertion.ContextualTuples,
+						}
+					}
 					resp, err := client.ListObjects(ctx, &pb.ListObjectsRequest{
 						StoreId:          storeID,
 						Type:             assertion.Request.Type,
 						Relation:         assertion.Request.Relation,
 						User:             assertion.Request.User,
-						ContextualTuples: assertion.Request.ContextualTuples,
+						ContextualTuples: contextualTuples,
 					})
 
 					if assertion.ErrorCode == 0 {
 						require.NoError(t, err)
-						require.ElementsMatch(t, assertion.Expectation, resp.Objects)
+						require.ElementsMatch(t, assertion.Expectation, resp.Objects,
+							fmt.Sprintf("ListObject(Type:%s,Relation:%s,User:%s,ContextualTuples:%v)\n",
```

Comment:
what is the `fmt.Sprintf("ListObject ...` doing? if it's a debug message, i'm pretty sure the standard output is enough

Reply:
Unfortunately it does not.
```
                    listobjects.go:130:
                        	Error Trace:	/Users/adriantam/src/github.com/openfga/openfga/tests/listobjects/listobjects.go:130
                        	Error:      	elements differ

                        	            	extra elements in list A:
                        	            	([]interface {}) (len=1) {
                        	            	 (string) (len=10) "document:1"
                        	            	}


                        	            	extra elements in list B:
                        	            	([]interface {}) (len=1) {
                        	            	 (string) (len=11) "document:1a"
                        	            	}


                        	            	listA:
                        	            	([]string) (len=1) {
                        	            	 (string) (len=10) "document:1"
                        	            	}


                        	            	listB:
                        	            	([]string) (len=1) {
                        	            	 (string) (len=11) "document:1a"
                        	            	}
                        	Test:       	TestListObjectsMySQL/RunAll/ListObjects/Schema1_1/this_with_contextual_tuples
```

I cannot tell what is the user/relation/object + any contextual tuples that cause the error.  It becomes problematic with more than 1 list object assertion in the test

Reply:
Before that error trace you should see the test name, once you have that it's easy to find the test data

Reply:
Yes, I see the name.  However, how can we tell which list object assertion fail if there are more than 1 assertion?

Reply:
Ah... in that case i think it's prettier to add `t.Logf("ListObject request: %s. Contextual tuples: %s", assertion.Request, assertion.ContextualTuples)` before the `require`. It will only print if you add `-v` to `go test`, or in case there was a failure.